bennu-core/src/main/java/org/fenixedu/bennu/core/security/Authenticate.java

```diff
@@ -50,26 +53,36 @@
      *            The response associated with the request that triggered the login
      * @param user
      *            The user to log in
+     * @param authenticationMethod
+     *            The authentication method used to login the user
      * @return
      *         The logged in user
      * @throws AuthorizationException
      *             If the provided user is {@code null} or if the user has its login expired
      */
-    public static User login(HttpServletRequest request, HttpServletResponse response, User user) {
+    public static User login(final HttpServletRequest request, final HttpServletResponse response, final User user, final String authenticationMethod) {
         if (user == null || user.isLoginExpired()) {
             throw AuthorizationException.authenticationFailed();
         }
 
-        HttpSession session = request.getSession();
-        loggedUser.set(user);
-        session.setAttribute(LOGGED_USER_ATTRIBUTE, user);
-        final Locale preferredLocale = user.getProfile().getPreferredLocale();
-        if (preferredLocale != null) {
-            I18NFilter.updateLocale(preferredLocale, request, response);
+        final AuthenticationContext authenticationContext = loggedUserContext.get();
+        if (authenticationContext == null) {
+            loggedUserContext.set(new AuthenticationContext(user, authenticationMethod));
+
+            storeInSession(request, authenticationContext);
```

Comment:
this will store always null. Is that the expected behaviour ?

Reply:
Fixed. Please check the commit below.